src/components/list/react/List.tsx

```diff
@@ -0,0 +1,201 @@
+interface IDefaultPropsType extends Partial<ListProps> {}
+
+/////////////////////////////
+//                         //
+//    Public attributes    //
+//                         //
+/////////////////////////////
+
+/**
+ * The display name of the component.
+ *
+ * @type {string}
+ * @constant
+ * @readonly
+ */
+const COMPONENT_NAME: string = `${COMPONENT_PREFIX}List`;
+
+/**
+ * The default class name and classes prefix for this component.
+ *
+ * @type {string}
+ * @constant
+ * @readonly
+ */
+const CLASSNAME: string = getRootClassName(COMPONENT_NAME);
+
+/**
+ * The default value of props.
+ *
+ * @type {IDefaultPropsType}
+ * @constant
+ * @readonly
+ */
+const DEFAULT_PROPS: IDefaultPropsType = {};
+/////////////////////////////
+
+/**
+ * List component - Use vertical layout to display elements
+ *
+ * @return {React.ReactElement} The component.
+ */
+const List: React.FC<ListProps> = ({
+    children,
+    className = '',
+    isClickable,
+    onListItemSelected,
+    ...props
+}: ListProps): React.ReactElement => {
+    // tslint:disable-next-line: typedef
+    const [activeItemIndex, setActiveItemIndex] = useState(-1);
+    // tslint:disable-next-line: typedef
+    const [hasItemSelected, setHasItemSelected] = useState(false);
+
+    /**
+     * Override the mouse down event - forward the event if needed
+     * @param {MouseEvent}  evt       Mouse event
+     * @param {number}      idx       Index of the target in the list
+     * @param {object}      itemProps Base props
+     */
+    // tslint:disable-next-line: typedef
+    const mouseDownHandler = (evt, idx, itemProps) => {
+        setActiveItemIndex(idx);
+        if (itemProps.onMouseDown) {
+            itemProps.onMouseDown(evt);
+        }
+    };
+
+    /**
+     * Handle keyboard interactions
+     * @param {KeyboardEvent}  evt Keybord input event
+     */
+    // tslint:disable-next-line: typedef
+    const onKeyInteraction = (evt: React.KeyboardEvent<HTMLUListElement>) => {
+        if (!isClickable) {
+            return;
+        }
+        if (evt.keyCode === DOWN_KEY_CODE) {
+            setActiveItemIndex(selectItemOnKeyDown(false));
+            evt.nativeEvent.preventDefault();
+            evt.nativeEvent.stopPropagation();
+        } else if (evt.keyCode === UP_KEY_CODE) {
+            setActiveItemIndex(selectItemOnKeyDown(true));
+            evt.nativeEvent.preventDefault();
+            evt.nativeEvent.stopPropagation();
+        } else if (evt.keyCode === TAB_KEY_CODE) {
+            evt.nativeEvent.preventDefault();
+            evt.nativeEvent.stopPropagation();
+        } else if (evt.keyCode === ENTER_KEY_CODE && onListItemSelected) {
+            evt.nativeEvent.preventDefault();
+            evt.nativeEvent.stopPropagation();
+            onListItemSelected(children[activeItemIndex]);
+        }
+    };
+
+    /**
+     * Find and select next or previous available list item.
+     * @param  {boolean}  previous Should the we look for previous item
```

Comment:
Did not understand ^^

Reply:
Definitely not clear :) I'll will change that